frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/SelectBox.jsx

```diff
@@ -71,149 +73,147 @@ function DataSourceSelect({
     }
   });
 
-  useEffect(() => {
-    function getForeignKeyDetails(incrementPageBy) {
-      if (isEmpty(searchValue)) {
-        const limit = 15;
-        const offset = (pageNumber - 1) * limit;
+  function setDefaultStateForSearch(makeSearchValueToDefault = false) {
+    setIsInitialForeignKeSearchDataLoaded(false);
+    setTotalSearchRecords(0);
+    setSearchPageNumber(1);
+    makeSearchValueToDefault && setSearchValue('');
+    setSearchResults([]);
+  }
+
+  function fetchForeignKeyDetails(page, totalRecords, isFirstPageLoaded, searchValue, foreignKeys, organizationId) {
+    const limit = 15;
+    const offset = (page - 1) * limit;
+
+    if (isFirstPageLoaded && offset >= totalRecords) return;
+    setIsLoadingFKDetails(true);
+    const referencedColumns = foreignKeys?.find((item) => item.column_names[0] === cellColumnName);
```

Comment:
can we handle foreign keys missing instead of optional chaining?

Reply:
Try to handle null and undefined cases explicitly instead of relying on optional chaining as these null/undefined values will trickle down and make it hard to reason about code IMHO. Only use them if we can't predict the optional part at all.